sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
```

Comment:
I certainly agree with the suggestion of making `checksum=True` default, but I don't support making the checking of existing files the default and only behavior. I use `sentinelsat` exactly as you described here: to run `download_all()` periodically with a full list of products against a folder with many existing files, expecting them to be skipped. This is useful if the server happened to be down during the last invocation, for example.
So, I suggest simply restoring the old behavior.